### PR TITLE
[WIP] Handle `yanked` flag

### LINKED ISSUE #146: The `yanked` flag should not be ignored
Currently `micropip.install()` ignores the `yanked` flag and even installs the yanked version.
For example, as of 2024-10-22, `micropip.install("python-multipart")` installs `0.0.13` even though [the version is yanked](https://pypi.org/project/python-multipart/0.0.13/).

CleanShot 2024-10-22 at 14 10 10@2x

```
>>> import micropip
>>> await micropip.install("python-multipart")
>>> import python_multipart
>>> python_multipart.__version__
'0.0.13'
```

---

As a reference, `pip` deals with the `yank` flag properly, as the yanked `python-multipart==0.0.13` is ignored as below for example.
```
.venv ❯ pip install python-multipart
Collecting python-multipart
  Obtaining dependency information for python-multipart from https://files.pythonhosted.org/packages/f5/0b/c316262244abea7481f95f1e91d7575f3dfcf6455d56d1ffe9839c582eb1/python_multipart-0.0.12-py3-none-any.whl.metadata
  Downloading python_multipart-0.0.12-py3-none-any.whl.metadata (1.9 kB)
Downloading python_multipart-0.0.12-py3-none-any.whl (23 kB)
Installing collected packages: python-multipart
Successfully installed python-multipart-0.0.12
```

### PR DESCRIPTION
**Pull request by whitphx**
Resolves #146

While this is still wip, let me create this PR to get reviews checking if it's in a right direction.

Could you check if this strategy is ok where I added `yanked` field to `WheelInfo` and use it to filter out the yanked wheels in `find_wheel`?

TODO:
* Write test
* Implement the recommended installer logic in PEP592: https://peps.python.org/pep-0592/#installers

### CODE REVIEW

File: micropip/package_index.py [hunk 1]
```diff
@@ -157,13 +157,16 @@ def _compatible_wheels(
                 # This key is not available in the Simple API HTML response, so this field may be None
                 size = file.get("size")
 
+                yanked = file.get("yanked")
```
Comment (hoodmane) [inline micropip/package_index.py:160]: I'd say you could just do:
```py
if file.get("yanked"):
    continue
```
here.

Comment (hoodmane) [reply micropip/package_index.py:160]: I guess this wouldn't allow for

> unless they are the only file that matches a version specifier that “pins” to an exact version using either == (without any modifiers that make it a range, such as .*) or ===. 

So maybe handling later is better.

Comment (whitphx) [reply micropip/package_index.py:160]: Yes I think that rule would be implemented in `find_wheel` so the `yanked` value should be kept in `WheelInfo` and handled there.

**Comment by hoodmane** [comment]
Why did you close this? Are you going to make another? If you add a test it looks like something we could merge.

**Comment by hoodmane** [comment]
Hmm I see your comment. Well thanks for looking into it @whitphx!

**Pull request closed without merging**